ruby/bindings/InitRubyBindings.cpp

```diff
@@ -920,6 +945,37 @@ void locateEmbeddedGems(bool use_bundler) {
 
     $logger.info "Bundling with groups [#{keep_groups.join(',')}]"
 
+    # Filter out dependencies with native extensions that we already have embedded in the CLI and that satisfy the requirements
+    locked_specs = Bundler.definition.instance_variable_get(:@locked_specs)
+
+    embedded_gems_to_activate.each do |spec|
+      next if spec.extensions.empty?
+
+      next unless locked_specs.any? {|locked_spec| locked_spec.name == spec.name}
+
+      dep_to_requirements = {}
+      locked_specs.each do |locked_spec|
+        deleted_deps = locked_spec.dependencies.select { |dep| dep.name == spec.name && dep.requirement.satisfied_by?(spec.version) }
+        locked_spec.dependencies.reject! { |dep| dep.name == spec.name && dep.requirement.satisfied_by?(spec.version) }
+        deleted_deps&.each do |dep|
+          dep_to_requirements[locked_spec.name] = dep.requirement.to_s
+        end
+      end
+
+
+      if locked_specs.none? { |locked_spec| locked_spec.dependencies.any? {|x| x.name == spec.name} }
+        locked_spec_v = locked_specs.select{|locked_spec| locked_spec.name == spec.name}.first.version
+        $logger.debug("Removing native gem #{spec.name} from Bundler locked_specs (version #{locked_spec_v}), using the embedded dependency (CLI one has version #{spec.version})")
+        $logger.trace("Gems having #{spec.name} as a dependency: #{dep_to_requirements}")
+        raise "Failed to activate #{spec.name}" unless spec.activate
+        $ignore_native_gem_names << spec.name
+        locked_specs.delete_by_name(spec.name)
+      end
+    end
```

Comment:
If after cleaning all dependencies, there are none left, then we actually remove the locked_spec called "spec.name" itself.

So for eg,

locked_specs has originally one for rubocop, with a dependency on json. We remove this one. And it was the only one.

Then we actually delete the locked_spec named json.

Reply:
As a result, when you run with Trace loglevel, you see


```
[ruby] <-1> Bundling with groups [default,development]
[ruby] <-2> Removing native gem json from Bundler locked_specs (version 2.7.2), using the embedded dependency (CLI one has version 2.6.3)
[ruby] <-3> Gems having json as a dependency: {"rubocop"=>"~> 2.3"}
Found no changes, using resolution from the lockfile
[ruby] <-1> Specs to be included [rake,public_suffix,addressable,ansi,ast,builder,multipart-post,faraday,minitar,rchardet,git,minitest,ruby-progressbar,minitest-reporters,parser,parallel,rainbow,regexp_parser,rexml,rubocop-ast,unicode-display_width,rubocop,rubocop-checkstyle_formatter,rubocop-performance,docile,simplecov-html,simplecov,openstudio_measure_tester,rubyzip,ruby-ole,spreadsheet,systemu,macaddr,uuid,yamler,zliby,bcl,bundler,diff-lcs,json-schema,matrix,sawyer,octokit,openstudio-ee,openstudio-extension,oslg,osut,topolys,tbd,openstudio-standards,openstudio-workflow,rspec-support,rspec-core,rspec-expectations,rspec-mocks,rspec]
```

Specs to be included no longer contains json.